---
title: "20201101-edgeR"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{R}
library(tidyverse)
library(edgeR)
```

Explore the effects of isogenic editing of APOE E4 to E3 in cerebral organoids. Comparison of APOE E4 vs E3 isogenic organoids with 3 biological replicates per group.

```{R}
### Read in counts
rawcounts<-read.table("../../output/counts/ExonCounts.txt", sep="\t", header=T, stringsAsFactors = F, quote="")

filtered.counts<-rawcounts

### Read in metadata
metadata<-read.table("../../data/SraRunTable.txt", sep=",", header = T, stringsAsFactors = F)
genotype<-factor(metadata$apoe_genotype, levels=c("E3E3_ISO", "E4E4_CTR"))

### Filtering genes with low expression
### cutoff - 10/L (L = min library size in millions)
10/(min(colSums((filtered.counts[,10:15])))/1000000)
### This will keep genes that have a count of at least 1 in at least 3 samples
keep <- rowSums(cpm(filtered.counts[,10:15], )>0.1)>=3 
table(keep)
#filtered.counts <-filtered.counts[keep, ]
#nrow(filtered.counts)

y <- DGEList(counts = filtered.counts[,10:15], genes = filtered.counts[,1:9], group = genotype)

keep <- filterByExpr(y, group=genotype)

table(keep)
### This is the same filtering as above

y <- y[keep, , keep.lib.sizes=FALSE]
y<-calcNormFactors(y)

### Write normalized counts
cpmout<-as.data.frame(cpm(y))
cpmout<-cbind(y$genes, cpmout)

write.table(cpmout, "../../output/counts/normalized-exon-counts.tsv", sep="\t", col.names = T, row.names = F, quote=F)
```

### Check sample clustering

```{R}
png("../../plots/edgeR/MDS.png", width = 800, height=600)
plotMDS(y, col=as.integer(factor(metadata$apoe_genotype)))
dev.off()
```

Samples show strong clustering based on APOE variant

### Get normalization factors for deepTools

```{R, eval=F}
NFs<-calcNormFactors(filtered.counts[,10:15], method = "TMM")

tmp.libsize<-colSums(filtered.counts[,10:15])

SizeFactors<-NFs*tmp.libsize / 1000000

SF.deepTools<-1/SizeFactors

outfile<-data.frame(SRR_ID=names(SF.deepTools), sizeFactors=SF.deepTools)

write.table(outfile, "../../output/counts/deeptools-exon-normalization-factors.txt", sep="\t", col.names = T, row.names = F, quote=F)
```



```{R}
###
design<-model.matrix(~genotype)

y<-estimateDisp(y, design)

fit <- glmFit(y, design)
colnames(design)

### Run exact test and get DE results
et<-exactTest(y)
et.genes<-topTags(et, n=Inf)$table

### Get significant genes with a fold change of 2 or more
significant<-et.genes[which(et.genes$FDR<0.05),]
significant.fc<-significant[which(abs(significant$logFC)>=1),]

### Write all genes and significant genes out to a file
write.table(et.genes, "../../output/DE/exons-APOE-E4-vs-APOE-E3.tsv", sep="\t", col.names = T, row.names = F, quote=F)
write.table(significant.fc, "../../output/DE/exons-APOE-E4-vs-APOE-E3-significant.tsv", sep="\t", col.names = T, row.names = F, quote=F)


### Alternatively, you can run a linear model and get DE results
#lrt <- glmLRT(fit,coef=2)
#DEtable <- topTags(lrt, n=Inf)$table

#significant<-DEtable[which(DEtable$FDR<0.05),]
#significant.fc<-significant[which(abs(significant$logFC)>=1),]
```

### Alternative splicing

```{R}
colnames(y$genes)
sp <- diffSpliceDGE(fit, coef=2, geneid="geneID", exonid="Start")

simes<-topSpliceDGE(sp, test="Simes", n=Inf)
de.simes<-simes[which(simes$FDR<0.05),]
write.table(de.simes, "../../output/splicing/differential-splicing-simestest.tsv", sep="\t", col.names = T, row.names = F, quote=F)

ftest<-topSpliceDGE(sp, test="gene", n=Inf)
de.ftest<-ftest[which(ftest$FDR<0.05),]
write.table(de.ftest, "../../output/splicing/differential-splicing-ftest.tsv", sep="\t", col.names = T, row.names = F, quote=F)

length(intersect(de.simes$geneID, de.ftest$geneID))
```



```{R}
png("../../plots/splicing/PCDHGC3-exonusage.png", width = 800, height = 600)
plotSpliceDGE(sp, geneid="PCDHGC3", genecol="Gene.name")
dev.off()

png("../../plots/splicing/PCDHGC4-exonusage.png", width = 800, height = 600)
plotSpliceDGE(sp, geneid="PCDHGC4", genecol="Gene.name")
dev.off()

plotSpliceDGE(sp, geneid="PCDHA6", genecol="Gene.name")

plotSpliceDGE(sp, geneid="SRSF5", genecol="Gene.name")

plotSpliceDGE(sp, geneid="NTRK2", genecol="Gene.name")

plotSpliceDGE(sp, geneid="TMEM161B-AS1", genecol="Gene.name")

plotSpliceDGE(sp, geneid="CHD4", genecol="Gene.name")

plotSpliceDGE(sp, geneid="PAX8-AS1", genecol="Gene.name")
```

